index.mjs

```diff
@@ -13,40 +14,38 @@ export default function icomesh(order = 4) {
         f, 0, -1, f, 0, 1, -f, 0, -1, -f, 0, 1
     ]);
 
-    let triangles = new Uint16Array([
+    let triangles = Uint16Array.of(
         0, 11, 5, 0, 5, 1, 0, 1, 7, 0, 7, 10, 0, 10, 11,
         11, 10, 2, 5, 11, 4, 1, 5, 9, 7, 1, 8, 10, 7, 6,
         3, 9, 4, 3, 4, 2, 3, 2, 6, 3, 6, 8, 3, 8, 9,
         9, 8, 1, 4, 9, 5, 2, 4, 11, 6, 2, 10, 8, 6, 7
-    ]);
-
-    let v = 12;
-    const midCache = order && new Map(); // midpoint vertices cache to avoid duplicating shared vertices
+    );
 
-    function addMidPoint(a, b) {
-        const key = Math.floor((a + b) * (a + b + 1) / 2) + (a < b ? a : b); // Cantor's pairing function
+    function addMidPoint(a, b, v) {
+        const key = ((a + b) * (a + b + 1) / 2) + Math.min(a, b) | 0; // Cantor's pairing function
         let i = midCache.get(key);
         if (i !== undefined) return i;
         midCache.set(key, v);
-        for (let k = 0; k < 3; k++) vertices[3 * v + k] = (vertices[3 * a + k] + vertices[3 * b + k]) / 2;
-        i = v++;
-        return i;
+        vertices[3 * v + 0] = (vertices[3 * a + 0] + vertices[3 * b + 0]) * 0.5;
+        vertices[3 * v + 1] = (vertices[3 * a + 1] + vertices[3 * b + 1]) * 0.5;
+        vertices[3 * v + 2] = (vertices[3 * a + 2] + vertices[3 * b + 2]) * 0.5;
+        return v;
     }
 
     let trianglesPrev = triangles;
+    const IndexArray = order > 5 ? Uint32Array : Uint16Array;
```

Comment:
What's the reason for moving this out of the loop? In theory, we would crunch less memory before because Uint16Array would be used for the first 5 orders even if we're targeting higher ones.

Reply:
Yes, but 16-bit registers slower than 32-bit (UInt32). Also this better for jit compiler. So I decided choose speed instead memory consumption